internal/rust/scanner/src/scanner.rl

```diff
@@ -128,14 +161,30 @@ int scan(int with_regex, const char **pp, const char *data, const char *pe, cons
     const char *ts;
     const char *te;
     unsigned int tok = ILLEGAL;
+    node_t* nls = NULL;
+    unsigned int no_nls = 0;
 
     %% write init nocs;
     %% write exec;
 
-    // Update output args
+    // Update output args.
     *token = tok;
     *token_start = ts - data;
     *token_end = te - data;
+    *newlines_len = no_nls;
+
+    // Now that the state machine has created a dynamic list of newline offsets (in reverse order),
+    // we can take that list and copy to an array for later use in Rust.
+    if (no_nls > 0) {
+        // We can now allocate contiguous memory given that we know the number of newlines found.
+        unsigned int *arr_nls = (unsigned int *) calloc(no_nls, sizeof(unsigned int));
```

Comment:
This call to `calloc` is of paramount importance instead of `int arr_nls[no_nls`. I should add a comment on that.